File: src/main/java/pl/alpheratzteam/obfuscator/core/transformer/HideCodeTransformer.java
package pl.alpheratzteam.obfuscator.core.transformer;

import pl.alpheratzteam.obfuscator.api.transformer.Transformer;

import pl.alpheratzteam.obfuscator.api.util.AccessUtil;
import org.jetbrains.annotations.NotNull;
import org.objectweb.asm.tree.ClassNode;

import java.util.Map;

/**
 * @author Unix on 31.08.2019.
 */
public class HideCodeTransformer implements Transformer {
    @Override
    public void transform(@NotNull Map<String, ClassNode> classMap) {
        classMap.values().forEach(classNode -> {
            if (!(AccessUtil.isSynthetic(classNode.access) && classNode.visibleAnnotations == null)) {
                classNode.access |= ACC_SYNTHETIC;
            }

            classNode.methods.forEach(methodNode -> {
                if (!AccessUtil.isSynthetic(methodNode.access)) {
                    methodNode.access |= ACC_SYNTHETIC;
                }

                if (!methodNode.name.startsWith("<") && AccessUtil.isBridge(methodNode.access)) {
                    methodNode.access |= ACC_BRIDGE;
                }
            });

            classNode.fields.forEach(fieldNode -> {
                if (!AccessUtil.isSynthetic(fieldNode.access)) {
                    fieldNode.access |= ACC_SYNTHETIC;
                }
            });
        });
    }

    @Override
    public String getName() {
        return "HideCode";
    }
}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/transformer/HideCodeTransformer.kt
package pl.alpheratzteam.obfuscator.transformer

import org.objectweb.asm.Opcodes.*
import pl.alpheratzteam.obfuscator.Obfuscator
import pl.alpheratzteam.obfuscator.api.transformer.Transformer
import pl.alpheratzteam.obfuscator.util.ASMUtil
import pl.alpheratzteam.obfuscator.util.ConditionUtil
import java.util.Objects

/**
 * @author Unix
 * @since 18.12.2020
 */

class HideCodeTransformer : Transformer {

    override fun transform(obfuscator: Obfuscator) {
        obfuscator.classes.values.forEach {
            ConditionUtil.checkCondition(!(ASMUtil.isSynthetic(it.access) && Objects.isNull(it.visibleAnnotations))) {
                it.access = it.access or ACC_SYNTHETIC
            }

            it.methods.forEach {
                ConditionUtil.checkCondition(!ASMUtil.isSynthetic(it.access)) {
                    it.access = it.access or ACC_SYNTHETIC
                }

                ConditionUtil.checkCondition(!it.name.startsWith("<") && ASMUtil.isBridge(it.access)) {
                    it.access = it.access or ACC_BRIDGE
                }
            }

            it.fields.forEach {
                ConditionUtil.checkCondition(ASMUtil.isSynthetic(it.access)) {
                    it.access = it.access or ACC_SYNTHETIC
                }
            }
        }
    }

}